### PR TITLE
Monotonicity checking for pMDPs and expected total rewards

### PR DESCRIPTION
**Pull request by jipspel**
- Implementation of monotonicity checking for reachability properties in pMDPs
- Implementation of monotonicity checking for reward properties in pMCs and pMDPs

### CODE REVIEW

**Comment by sjunges** [comment]
We should squash this PR rather than merge it, given that it contains >500 commits ;-)

**Comment by sjunges** [comment]
The test folder contains an awful lot of changes that I do not understand, in particular, quite some new files...

File: src/storm-pomdp-cli/storm-pomdp.cpp [hunk 1]
```diff
@@ -358,6 +358,7 @@ namespace storm {
                     std::string transformMode = transformSettings.getFscApplicationTypeString();
                     auto pmc = toPMCTransformer.transform(storm::transformer::parsePomdpFscApplicationMode(transformMode));
                     STORM_PRINT_AND_LOG(" done.\n");
+
```
Comment (sjunges) [inline src/storm-pomdp-cli/storm-pomdp.cpp:361]: Please undo this change as this file should not be affected by an extension to parameter lifting

File: src/storm-pars/storage/ParameterRegion.cpp [hunk 2]
```diff
@@ -144,6 +182,38 @@ namespace storm {
...
+                result.insert(typename Valuation::value_type(variable, (this->getLowerBoundary(variable) + this->getUpperBoundary(variable))/2));
+            }
+            return result;
+        }
+
+        template<typename ParametricType>
+        typename ParameterRegion<ParametricType>::Valuation ParameterRegion<ParametricType>::getSplittingPoint(const std::set<VariableType>& consideredVariables, const std::set<VariableType>& possiblyMonotoneIncrVars, const std::set<VariableType>& possiblyMonotoneDecrVars, bool minimize) const {
+            Valuation result;
+            for (auto const& variable : consideredVariables) {
+//                if (possiblyMonotoneIncrVars.find(variable) != possiblyMonotoneIncrVars.end()) {
```
Comment (sjunges) [inline src/storm-pars/storage/ParameterRegion.cpp:199]: something is broken here?

File: src/storm/storage/SparseMatrix.h [hunk 3]
```diff
@@ -632,6 +632,11 @@ class SparseMatrix {
      */
     void makeRowGroupingTrivial();
 
+    /*!
+     * Creates a new square matrix by merging the entries in a rowgroup (adding the entries)
+     */
+    SparseMatrix getSquareMatrix();
```
Comment (sjunges) [inline src/storm/storage/SparseMatrix.h:638]: I find the name getSquareMatrix very confusing... :-)

Comment (jipspel) [reply src/storm/storage/SparseMatrix.h:638]: Do you have a suggestion for the name?

**Comment by sjunges** [comment]
We observe that we have conflicts and see that pMDPs and monotonicity is not a huge step forward. We therefore will not merge this in the foreseaable future.
